src/main/java/games/strategy/net/nio/Encoder.java

```diff
@@ -37,21 +37,20 @@ void write(final SocketChannel to, final MessageHeader header) {
     if (to == null) {
       throw new IllegalArgumentException("No to channel!");
     }
-    final ByteArrayOutputStream2 sink = new ByteArrayOutputStream2(512);
-    SocketWriteData data;
     try {
+      final ByteArrayOutputStream sink = new ByteArrayOutputStream(512);
       write(header, objectStreamFactory.create(sink), to);
-      data = new SocketWriteData(sink.getBuffer(), sink.size());
+      final SocketWriteData data = new SocketWriteData(sink.toByteArray(), sink.size());
+      if (logger.isLoggable(Level.FINER)) {
+        logger.log(Level.FINER, "encoded  msg:" + header.getMessage() + " size:" + data.size());
+      }
+      writer.enque(data, to);
     } catch (final Exception e) {
       // we arent doing any io, just writing in memory
       // so something is very wrong
       logger.log(Level.SEVERE, "Error writing object:" + header, e);
       return;
     }
-    if (logger.isLoggable(Level.FINER)) {
-      logger.log(Level.FINER, "encoded  msg:" + header.getMessage() + " size:" + data.size());
-    }
-    writer.enque(data, to);
```

Comment:
My only concern is that by moving this code into the `try` block, we blur the original author's intention of swallowing the `IOException`s that may be thrown on line 42.  Now if any other exception is thrown for an unrelated reason (e.g. someone introduces an NPE in `NioWriter#enqueue()`), we will keep plowing ahead when we should have failed fast.

If we're going to put everything in this `try` block, I would feel more comfortable if we only catch the exception we know means "just writing in memory, so something is very wrong," namely `IOException`.  In fact, there's an unchecked exception thrown by `SocketWriteData#SocketWriteData()` that we probably shouldn't be swallowing (that's a legacy issue that existed prior to this change).

Reply:
Good catch @ssoloff I changed this catch block to only catch IOExeptions